homeassistant/components/xiaomi_miio/__init__.py

```diff
@@ -75,6 +85,50 @@ def get_platforms(config_entry):
     return []
 
 
+@callback
+def create_miio_device_and_coordinator(
+    hass: core.HomeAssistant, entry: config_entries.ConfigEntry
+):
+    """Set up a data coordinator and one mio device to service multiple entities."""
+    model = entry.data[CONF_MODEL]
+    host = entry.data[CONF_HOST]
+    token = entry.data[CONF_TOKEN]
+    name = entry.title
+    device = None
+    if model not in MODELS_HUMIDIFIER:
+        return
+    if model in MODELS_HUMIDIFIER_MIOT:
+        device = AirHumidifierMiot(host, token)
+    else:
+        device = AirHumidifier(host, token, model=model)
+
+    async def async_update_data():
+        """Fetch data from the device using async_add_executor_job."""
+        # On state change the device doesn't provide the new state immediately.
+        try:
+            async with async_timeout.timeout(10):
+                return await hass.async_add_executor_job(device.status)
+
+        except DeviceException as ex:
+            _LOGGER.error("Got exception while fetching the state: %s", ex)
+
+    # Create update miio device and coordinator
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        # Name of the data. For logging purposes.
+        name=name,
+        update_method=async_update_data,
+        # Polling interval. Will only be polled if there are subscribers.
+        update_interval=timedelta(seconds=10),
+    )
+
```

Comment:
Should we do the first data fetch (`coordinator.async_refresh`) here already, instead of in the platforms?

Reply:
async_request_refresh is only needed to trigger an update after a change was made to the device, Every 10 sec the device is polled and updated in the platforms already. I have added an initial data fetch now, it requires to await the function because triggering should be awaited.